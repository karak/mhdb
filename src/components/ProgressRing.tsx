import * as React from 'react';
import * as ReactDOM from 'react-dom';
import RefreshIndicator from 'material-ui/RefreshIndicator';

/** ProgressRing Props */
interface ProgressRingProps {
  /** flag to show or hide */
  loading?: boolean;
}

// tslint:disable-next-line:function-name
export default function ProgressRing({ loading = true }: ProgressRingProps) {
  return (
    <RefreshIndicator left={0} top={0} status={loading ? 'loading' : 'hide'} />
  );
}
