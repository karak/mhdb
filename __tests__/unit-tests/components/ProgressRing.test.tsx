import '../../helpers/setup';
import * as React from 'react';
import ProgressRing from '../../../src/components/ProgressRing';
import { shallow } from 'enzyme';
import { default as toJson } from 'enzyme-to-json';

describe('ProgressRing', () => {
  describe('props loading', () => {
    it('is visible if true', async () => {
      const wrapper = shallow(<ProgressRing loading={true} />);
      const deepWrapper = wrapper.find('RefreshIndicator');

      expect(deepWrapper.prop('status')).toBe('loading');
    });

    it('is visible by default', async () => {
      const wrapper = shallow(<ProgressRing />);
      const deepWrapper = wrapper.find('RefreshIndicator');

      expect(deepWrapper.prop('status')).toBe('loading');
    });

    it('is hidden if false', async () => {
      const wrapper = shallow(<ProgressRing loading={false} />);
      const deepWrapper = wrapper.find('RefreshIndicator');

      expect(deepWrapper.prop('status')).toBe('hide');
    });
  });

  it('matches with snapshot', () => {
    const tree = toJson(shallow(<ProgressRing />));

    expect(tree).toMatchSnapshot();
  });
});
