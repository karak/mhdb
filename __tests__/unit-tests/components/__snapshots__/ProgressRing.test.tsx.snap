// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressRing matches with snapshot 1`] = `
<RefreshIndicator
  left={0}
  percentage={0}
  size={40}
  status="loading"
  top={0}
/>
`;
